### PR TITLE
deconstruction with RCD returns charges

### PR DESCRIPTION
**Pull request by yglop**
## About the PR
read the title.

## Why / Balance
because???

## Technical details
simple if else check

## Media
RCD.png

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

**Changelog**
:cl:
- tweak:  Deconstruction with RCD returns charges.

## Summary by CodeRabbit

- **New Features**
	- Introduced the `MaterialChargeComponent` for managing a whitelist of materials.
	- Added the `MaterialChargeSystem` to facilitate interaction between materials and charge management.
	- New `MaterialCharge` entity allows interaction with a variety of materials, enhancing tool functionality.
	- Updated RCD functionality to add charges when in deconstruction mode.

These updates improve gameplay by enabling more dynamic material interactions and charge management.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
media?

**Comment by Piras314** [comment]
being able to make windows with steel feels kind of weird

**Comment by Scruq445** [comment]
any engineer mained after 1993 cant craft, all they know is deconstruct captain door, make frezon, steal from CE, get rad poisoning, charge they RCD and die.

**Comment by yglop** [comment]
> being able to make windows with steel feels kind of weird

you can do that in SS13 and i dont want to rewrite whole RCD system (upstream will brake it one day).

**Comment by yglop** [comment]
> media?

media of what? me clicking with a steel sheet on RCD and steel sheet disappearing?

**Comment by JoeHammad1844** [comment]
> media of what? me clicking with a steel sheet on RCD and steel sheet disappearing?

yeah

**Comment by whateverusername0** [comment]
@yglop what if, instead of this, RCD gained material when destroying something?

**Comment by yglop** [comment]
> @yglop what if, instead of this, RCD gained material when destroying something?

done

File: Content.Shared/RCD/Systems/RCDSystem.cs [hunk 1]
```diff
@@ -272,7 +272,11 @@ private void OnDoAfter(EntityUid uid, RCDComponent component, RCDDoAfterEvent ar
 
         // Play audio and consume charges
         _audio.PlayPredicted(component.SuccessSound, uid, args.User);
-        _charges.UseCharges(uid, args.Cost);
+        // Goobstation - start
+        if (component.CachedPrototype.Mode == RcdMode.Deconstruct)
+            _charges.AddCharges(uid, args.Cost / 2);
+        else _charges.UseCharges(uid, args.Cost);
+        // Goobstation - end
```
[the remark this answers is not in the document]
Comment (yglop) [reply Content.Shared/RCD/Systems/RCDSystem.cs:279]: нет.

**Review by whateverusername0 — APPROVED** [review]

File: Content.Shared/RCD/Systems/RCDSystem.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (TurboTrackerss14) [reply Content.Shared/RCD/Systems/RCDSystem.cs:279]: Me when I don't make unit tests or documentation and accidentally create RCD ammo duplication. thank you very cool.

**Comment by gusxyz** [comment]
please revert this

**Pull request merged**